functions.md

```diff
@@ -0,0 +1,131 @@
+## Function
+A function is a reuseable block of code, that group together a sequence of statement to peform a specific task.
+### Function declaration
+in javaScript they are many ways to declare a function.One way to create a function is by using a `function` declaration. example:
+
+```js
+function teamNames() {
+    console.log('Liverpool', 'Barcelona');
+}
+console.log(teamNames()); // Liverpool, Barcelona
+```
+A function declaration consists of:
+
+- The function keyword.
+- The name of the function, or its identifier, followed by parentheses.
+- A function body, or the block of statements required to perform a specific task, enclosed in the function’s curly brackets, { }.
+
+### Calling a function
+A function is called when `function` name followed by the parentheses is typed. From the example above
+
+```js
+console.log(teamNames());
+```
+called the function inside `teamNames`.
+
+### Parameter and Argument
+parameter allows function accept input(s), and perform a task using the input(s). When declaring a function we can specify a parameter.  We use parameters as placeholders for information that will be passed to the function when it is called. example:
+
+```js
+function aggregateScore(Home, Away) {
+    console.log(Home + Away);
+}
+```
+the value passed to the functiion when they are called, are called `argument`. it can be passed as a value or variable example:
+
+```js
+function aggregateScore(Home, Away) {
+    console.log(Home + Away);
+}
+aggregateScore(4 + 2)// 6
+```
+
+### Default parameters
+in `ES6` default parameter feature is included, with the ability to have a predetermined value, in case  there is no argument passed into function or the argument is `undefined` when called. example:
+
+```js
+function greeting (name = 'stranger') {
+  console.log(`Hello, ${name}!`)
+}
+
+greeting('Nick') // Output: Hello, Nick!
+greeting() // Output: Hello, stranger!
+```
+By using a default parameter, we account for situations when an argument isn’t passed into a function that is expecting an argument.
+
+### Return
+when a function is called, the computer will run through the functions code and evaluate the result of calling the function. By default that resulting value is undefined. example:
+
+```js
+function monitorCount(rows, columns) {
+   let total = rows + columns
+}
+console.log(monitorCount(3 + 5));// print undefined
+```
+
+ the `return` keyword allow us see the calculation that the computer did print example:
+
+```js 
+function totalNumber(cash, bank) {
+    return 30 +80;
+}
+console.log(totalNumber()); // 11
+```
+
+The `return` keyword is powerful because it allows functions to produce an output. We can then save the output to a variable for later use
+
+### Function Expression
+ Another way of defining a function is to use a function expression. example:
+ 
+ ```js
+ const myName = Function(boy {
+
+ })
```

Comment:
Investigate this properly, this may not be the right way to declare a [function expression](https://developer.mozilla.org/en-US/docs/web/JavaScript/Reference/Operators/function)